### PR TITLE
Update the node version and GH actions in workflows

### PR DESCRIPTION
**Pull request by broccolinisoup**
Closes https://github.com/primer/octicons/issues/946

Octicons are failed to be publish to NPM due to an [error](https://github.com/primer/octicons/actions/runs/4919743480/jobs/8793883454) while fetching the docker image to build nodes. 

Apparently Debian 9/stretch moved to archive.debian.org ([reference](https://lists.debian.org/debian-devel-announce/2023/03/msg00006.html))

Found through this thread: https://unix.stackexchange.com/questions/743839/apt-get-update-failed-to-fetch-debian-amd64-packages-while-building-dockerfile-f

Also updating all other node versions and GH actions.

### CODE REVIEW

File: .github/actions/build_node/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:10-slim
+FROM node:20-slim
```
Comment (broccolinisoup) [inline .github/actions/build_node/Dockerfile:1]: https://hub.docker.com/_/node/

I wasn't sure which one to change to 🙈 Feel free to update if there is an image that suits/works better!

[the remark this answers is not in the document]
Comment (colebemis) [reply .github/actions/build_node/Dockerfile:1]: I think we should also update the `node-version` in these workflows:

https://github.com/primer/octicons/blob/main/.github/workflows/publish.yml#L12

https://github.com/primer/octicons/blob/4f0c6b4cff47e3363b212fc3a843ac8e775ca4b0/.github/workflows/release.yml#L24

https://github.com/primer/octicons/blob/4f0c6b4cff47e3363b212fc3a843ac8e775ca4b0/.github/workflows/ci.yml#L15

[the remark this answers is not in the document]
Comment (broccolinisoup) [reply .github/actions/build_node/Dockerfile]: Thanks @colebemis! I updated all node versions and also the github actions (like checkout, setup-node etc) from v2 to v3

**Review by colebemis — APPROVED** [review]
Looks great 🎉

**Pull request merged**